### DISCUSSION TITLE
How to make a non-Sizable form with Minimise button enabled just like in VB6?

### DISCUSSION THREAD

**Discussion by usman-vb** [twinbasic/twinbasic#1707] [Q&A] [answered]
First of all, thanks a lot for providing twinBASIC, the Open Source Version of VB6. I have tried to create some basic programs in twinBASIC and also converted one of my VB6 programs into twinBASIC and it worked very well but found a simple problem.

In VB6, if we want a form having a fixed Height and Width with Minimise button enabled then we will take these steps:
1.	Start a New project. Select the form and then select BorderStyle: Fixed Single
2.	Select MinButton=True

This will disable only Max Button that is OK because we only want a non-Resizable Form. But in twinBASIC, if BorderStyle: Fixed Single is selected, then the both Minimise and Maximise buttons will be gone (even if we again select MinButton=True). How to make a non-Sizable form with Minimise button enabled?
I’m using IDE 416 Beta version.
I have found that it is changed in version 411 (October 20, 2023):
fixed: (regression) Form.BorderStyle of vbFixedSingle was showing min/max buttons [ https://github.com/twinbasic/twinbasic/issues/1268]
Thanks.

**Comment by fafalone** [comment, marked as answer]
twinBASIC is currently in a pre-release, Beta stage, so while it's getting so very close to fully matching VB6, there's still some issues like this one. You *should* be able to do it by the same settings you use for VB6, and that you currently can't means this would be fall under either unimplemented feature or a bug. 

Thanks for reporting the issue, I'm sure Wayne (the developer of twinBASIC) will get this fixed, though I couldn't tell you when. So much to do, sadly only one of him 😄 

----

As a temporary workaround until this is fixed properly, you can follow this procedure:

Add the following declarations to your form:

```
Private Enum GWL_INDEX
    GWL_WNDPROC = (-4)
    GWL_HINSTANCE = (-6)
    GWL_HWNDPARENT = (-8)
    GWL_ID = (-12)
    GWL_STYLE = (-16)
    GWL_EXSTYLE = (-20)
    GWL_USERDATA = (-21)
End Enum
#If Win64 Then
Private DeclareWide PtrSafe Function GetWindowLong Lib "user32" Alias "GetWindowLongPtrW" (ByVal hWnd As LongPtr, ByVal nIndex As GWL_Index) As LongPtr
Private DeclareWide PtrSafe Function SetWindowLong Lib "user32" Alias "SetWindowLongPtrW" (ByVal hWnd As LongPtr, ByVal nIndex As GWL_Index, ByVal dwNewLong As LongPtr) As LongPtr
#Else
Private DeclareWide Function GetWindowLong Lib "user32" Alias "GetWindowLongW" (ByVal hWnd As Long, ByVal nIndex As GWL_INDEX) As Long
Private DeclareWide Function SetWindowLong Lib "user32" Alias "SetWindowLongW" (ByVal hwnd As Long, ByVal nIndex As GWL_INDEX, ByVal dwNewLong As Long) As Long
#End If

Private Const WS_MINIMIZEBOX = &H00020000
```

Then add the following to your `Form_Load` code:

```
        Dim dwStyle As Long
        dwStyle = CLng(GetWindowLong(Me.hWnd, GWL_STYLE))
        dwStyle = dwStyle Or WS_MINIMIZEBOX
        SetWindowLong Me.hWnd, GWL_STYLE, dwStyle
```

This solution is compatible with both 32bit and 64bit programs.

**Reply by usman-vb** [reply]
It has solved the problem, getting back the minimize button.  Thank you so much for your help and providing instant solution.
